### ISSUE TITLE
How to use this model to detect a new sentence?

### ISSUE THREAD

**Issue by shallow777** [YichenZW/Coh-MGT-Detection#3]
I have a trained model already. How can i use this model to detect a sentence created by which model?I am looking forward to your reply. Thanks a lot.

**Comment by YichenZW** [comment]
Hi shallow777, thank you for your question! It is possible, but this code is written for binary classification (i.e., whether a text is human-written or machine-generated), so maybe you need to revise some lines, e.g., the output logits length (run_detector.py#L1135), to do the multi-class classification.

**Comment by shallow777** [comment]
sorry, what i said is not quite accurate. I am wondering whether there is a example of code to detect the new sentence(maybe just a paragraph or a short article). Thanks a lot.

**Comment by YichenZW** [comment]
oh, gotta. If you mean "new sentence" that is not from the existing dataset, it is possible. I can add a script in the README in the next revision. Is that what you are wondering?

**Comment by shallow777** [comment]
yes, it is exactly what i mean. Thanks a lot for your help.

**Comment by LZ-Dong** [comment]
Hi YichenZW，I have a similar question. I have trained a gpt2 model with default parameters. In post hoc evaluation (run_detector.py#L1392 #L1405) something went wrong.
![img1](https://github.com/YichenZW/Coh-MGT-Detection/assets/104953644/e3a9ec44-6386-449a-b84c-d0f9df4231b1)
I simply changed `AutoModelForSequenceClassification.from_pretrained` to `RobertaForGraphBasedSequenceClassification_RFCL.from_pretrained`, but the post hoc evaluation gave the result that f1 = 0.
![img2](https://github.com/YichenZW/Coh-MGT-Detection/assets/104953644/46b614d0-7834-44d9-8c27-0a06173ffc55)
How can I run the post hoc evaluation correctly? I am looking forward to your reply. Thanks a lot.
